float64 mass
float64 COM
---
bool success
